runtime/autoload/fswatch.vim

```diff
@@ -0,0 +1,46 @@
+if exists('g:loaded_watcher_provider')
+  finish
+endif
+
+let s:save_cpo = &cpo " save user coptions
+set cpo&vim " reset them to vim defaults
+
+command! -nargs=1 Watch call v:lua.vim.fswatch.start_watch(expand('<args>'))
+command! -nargs=1 Stop call v:lua.vim.fswatch.start_watch(expand('<args>'))
+
+" function to prompt the user for a reload
+function! fswatch#PromptReload(buf)
+  let choice = confirm("File ".bufname(a:buf)." changed. Would you like to reload?","&Yes\n&Show diff\n&No", 1)
+  if choice == 1
+    call fswatch#Reload(a:buf)
+  elseif choice == 2
+    let s:save_swb = &switchbuf
+    execute 'set switchbuf=usetab'
+    execute 'sb '.a:buf
+    call fswatch#DiffOrig()
+    execute 'set switchbuf='.s:save_swb
+    unlet s:save_swb
+  endif
+endfunction
+
+" function to reload the buffer
+function! fswatch#Reload(buf)
+  execute 'silent checktime '.a:buf
+endfunction
+
+" function display the diff between the current buffer and original file
+function! fswatch#DiffOrig()
```

Comment:
This would require a bit more than just lua work. The problems seems to be with the fact that `:b` tries reloading files. (Maybe it calls `checktime`?) So if I open a new tab for this, and call `:b` to get the buffer, the buffer will be updated and there will be no diff to show. If I turn off autoread it'll be a second prompt like `checktime`.
As per the discussion, I will take a look at the C code and see how `:b` works, and if we need to refactor `:checktime` a bit.

Reply:
Done via some C code.